hathor/manager.py

```diff
@@ -977,14 +972,18 @@ def on_new_tx(
         :param fails_silently: if False will raise an exception when tx cannot be added
         :param propagate_to_peers: if True will relay the tx to other peers if it is accepted
         """
-        return self.vertex_handler.on_new_vertex(
+        success = self.vertex_handler.on_new_vertex(
             tx,
             quiet=quiet,
             fails_silently=fails_silently,
-            propagate_to_peers=propagate_to_peers,
             reject_locked_reward=reject_locked_reward,
         )
 
+        if propagate_to_peers and success:
```

Comment:
I've seen this same block of code over and over in this PR. I'm wondering how we can remove this duplicated code.

Reply:
This code is duplicate in the following places:

- 2 times in sync-v1 agent.
- 1 time in sync-v2 agent.
- 1 time in sync-v2 mempool.
- 1 time in `HathorManager.on_new_tx`.

Considering sync-v1 will be deprecated and removed soon, and `HathorManager.on_new_tx` will also be removed, we'll be left with only two occurrences of this simple `if`. I would rather keep it like this for now, even if temporarily, because my next PRs need this dependency removed — sync-v2 will no longer call `ConnectionManager.send_tx_to_peers()`, and instead will perform this though IPC.

After those refactors are complete, we may remove this code and improve it by making the `ConnectionsManager` read the PubSub message for when a new tx is accepted, and propagate it automatically to the interested peers, as suggested by you.

I created an issue so we can do that in the future: https://github.com/HathorNetwork/hathor-core/issues/1168

Reply:
Ok. Thanks for the explanation and for creating this issue to refactor it later.